File: chaerim/Programmers/Level_0/Lv0_다항식_더하기.py
def solution(polynomial):
    answer = ''
    polynomial = polynomial.split(" + ")

    num_count = 0
    x_count = 0
    for term in polynomial:
        if 'x' in term:
            if term == 'x':
                x_count += 1
            else:
                x_count += int(term[:-1])
        else:
            num_count += int(term)
    
    if num_count == 0:
        answer = str(x_count) + 'x'
        if x_count == 1:
            answer = 'x'
    elif x_count == 0:
        answer = str(num_count)
    elif x_count == 1:
        answer = 'x + ' + str(num_count)
    else:
        answer = str(x_count) + 'x + ' + str(num_count)

    return answer

File: seongwon/Programmers/Lv0_다항식_더하기.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cctype>
using namespace std;

string solution(string polynomial) {
    string answer = "";
    vector<string> str;
    int index = 0;
    for(int i = 0; i<polynomial.size(); i++){
        if(polynomial[i] == ' '){
            str.push_back(polynomial.substr(index, i-index));
            index = i+1;
        }
    }
    str.push_back(polynomial.substr(index, polynomial.size()));
    int x = 0;
    int c = 0;
    
   for(string s : str){
    if(s == "+" || s.empty()){ // "+" 문자열 , 공백 무시
        continue; 
    }
    else if(s == "x"){
        x++;    
    }
    else if(s.back() == 'x'){
        x += stoi(s);
    }
    else{
        c += stoi(s);
    }
}


    
    if(x!=0){
        if(x==1)    answer += "x";  //x가 1이면 "1x"가 아닌 "x"를 반환해야함
        else answer += to_string(x)+"x";
        if(c!=0){
            answer += " + ";
            answer += to_string(c);
        }
    }
    else{
        answer += to_string(c);
    }
    
    return answer;
}

/*
stoi => 문자열을 정수로 반환 "234b" -> 234
.back() => 가장 뒤쪽원소 반환
*/
